hack/tools.go

```diff
@@ -0,0 +1,23 @@
+// +build tools
```

Comment:
why is this file needed ?

Reply:
I see, its that you can keep the depedency in go.mod on goimport so that it can be vendored and is not accidentally removed.

It still feels very hackish to me and is a clear smell, that dev dependencies should be managed separately (but it looks like its a go limitation). 

I could see two solutions here:

* Don't compile devtools like "goimports" on the fly but use a given binary (like from a container image)
* Use an own kind of vendor by committing dev dependencies in a different vendor dire (vendor_dev) and fiddle with GOPATH before calling `go get goimports`. 

I agree that using goimports as it is now is not the correct way, also because the `go get` call will update `modules.txt` which sometimes leads to a diff to the comitted modules.txt when running in the CI. Up to now fixed the modules.txt manually in such a case, but vendoring seems to be the better solution here.

Reply:
If this does not work out or is too involved, than we can of course vendor `goimports`.

But we should keep `-mod=vendor` for all go commands everywhere to be explicit and not relying on defaults which can change (which seems to happen quite a bit in go-land as just seen when going from go 1.13 to 1.14)